### PR TITLE
Mfeigl/add scene properties expressions

### PR DESCRIPTION
**Pull request by mfeigl**
Added new sample, called Scene Properties Expressions. The scene contains a cone, and its heading and pitch can be modified via QSliders.

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Scene Properties Expressions
+
+This sample demonstrates how to update the orientation of a graphic using scene property rotation expressions via QSlider.
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/README.md]: I don't think the type is called `QSlider` for qt quick. Also, the point of the sample is not that we use a slider - just that we update the properties, so how about just:

> This sample demonstrates how to update the orientation of a graphic using scene property rotation expressions.

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+![](ScenePropertiesExpressions.gif)
+
+## How to use the sample
+Move the `Heading` and `Pitch` sliders to change the orientation of graphics in the graphics overlay.
+
+## How it works
+The sample creates a GraphicOverlay and a SimpleRenderer. The SimpleRenderer has expressions for `HEADING` and `PITCH`, which are set via `simpleRenderer.setHeadingExpression("HEADING");`. The simple renderer is set to the graphics overlay via `graphicsOverlay->setRenderer(simpleRenderer);`. A graphic is created and added it to the graphics overlay. The graphic's rotation is set via `Graphic.getAttributes.put("HEADING", heading)` where the attribute key is the expression and the value is the rotation angle.
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/README.md]: type `GraphicOverlay` => `GraphicsOverlay`

Comment (lsmallwood) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/README.md]: > set via `simpleRenderer.setHeadingExpression("HEADING");

shouldn't that be "set via `simpleRenderer.setHeadingExpression("[HEADING]");" ?

Comment (lsmallwood) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/README.md]: "added it to" => "added to"

Comment (lsmallwood) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/README.md]: `Graphic.getAttributes.put("HEADING", heading)`

this isn't correct for the Qt API

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.cpp [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+#include "SceneQuickView.h"
+#include "SimpleMarkerSceneSymbol.h"
+#include "SimpleRenderer.h"
+
+using namespace Esri::ArcGISRuntime;
+
+namespace
+{
+  const QString HEADING("HEADING");
+  const QString PITCH("PITCH");
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.cpp]: 👍

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.cpp [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        QUrl("https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"), this);
+
+  // add the elevation source to the scene to display elevation
+  m_scene->baseSurface()->elevationSources()->append(elevationSource);
+
+  m_graphicsOverlay = new GraphicsOverlay(this);
+  m_graphicsOverlay->setSceneProperties(LayerSceneProperties(SurfacePlacement::Relative));
+
+  // Create a SimpleRenderer and set expressions on its scene properties:
+  // Renderer.getSceneProperties().setHeadingExpression("[HEADING]").
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.cpp]: I don't think you need this line - it's just duplicating the code?

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.cpp [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  // create a camera
+  const double latitude = 32.09;
+  const double longitude = -118.71;
+  const double altitude = 100000.0;
+  const double heading = 0.0;
+  const double pitch = 45.0;
+  const double roll = 0.0;
+  const int coneDimension = 10000;
+
+  Camera camera(latitude - 1.0, // place the camera arbitrarily southern than the cone
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.cpp]: > // place the camera arbitrarily southern than the cone

should be:

> // place the camera arbitrarily south of the cone

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.cpp [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+                longitude,
+                altitude * 2, // place the camera arbitrarily higher than the cone
+                heading,
+                pitch,
+                roll);
+
+  // set the viewpoint
+  m_sceneView->setViewpointCameraAndWait(camera);
+
+
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.cpp]: remove empty line

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.h [hunk 7]
```diff
@@ -0,0 +1,66 @@
+// [WriteFile Name=ScenePropertiesExpressions, Category=Scenes]
+// [Legal]
+// Copyright 2018 Esri.
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.h]: 2019

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.pro [hunk 8]
```diff
@@ -0,0 +1,63 @@
+#-------------------------------------------------
+# Copyright 2018 Esri.
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.pro]: 2019 (throughout)

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qml [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+            top: parent.top
+        }
+
+        height: childrenRect.height
+
+        Text {
+            anchors {
+                margins: 5
+            }
+            objectName: "pitch"
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qml]: don't need `objectName`

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qrc [hunk 10]
```diff
@@ -0,0 +1,11 @@
+<RCC>
+    <qresource prefix="/Samples/Scenes/ScenePropertiesExpressions">
+        <file alias="sample.json">ScenePropertiesExpressions.json</file>        
+        <file>ScenePropertiesExpressions.qml</file>
+        <file>ScenePropertiesExpressions.h</file>
+        <file>ScenePropertiesExpressions.cpp</file>
+		<file>main.qml</file>
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qrc]: indentation

File: ArcGISRuntimeSDKQt_QMLSamples/Scenes/ScenePropertiesExpressions/README.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
+# Scene Properties Expressions
+
+This sample demonstrates how to update the orientation of a graphic using scene property rotation expressions via QSlider.
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_QMLSamples/Scenes/ScenePropertiesExpressions/README.md]: same comments as cpp README

File: ArcGISRuntimeSDKQt_QMLSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qml [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+                    style: Enums.SimpleMarkerSceneSymbolStyleCone
+                    color: "red"
+                    width: coneDimension
+                    depth: coneDimension
+                    height: coneDimension * 2
+                    anchorPosition: Enums.SceneSymbolAnchorPositionCenter
+                }
+
+                Component.onCompleted: {
+                    coneGraphic.attributes.insertAttribute("Heading", initialHeading);
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_QMLSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qml]: you could centralize the attribute names as a property

File: ArcGISRuntimeSDKQt_QMLSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qml [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+            top: parent.top
+        }
+
+        height: childrenRect.height
+
+        Text {
+            anchors {
+                margins: 5
+            }
+            objectName: "pitch"
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_QMLSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qml]: don't need objectName

File: ArcGISRuntimeSDKQt_QMLSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qml [hunk 14]
```diff
@@ -0,0 +1,184 @@
...
+            onValueChanged: {
+                coneGraphic.attributes.replaceAttribute("Pitch", value);
+            }
+        }
+
+        Text {
+            anchors {
+                margins: 5
+            }
+            objectName: "heading"
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_QMLSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qml]: don't need objectName

File: ArcGISRuntimeSDKQt_QMLSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qml [hunk 15]
```diff
@@ -0,0 +1,184 @@
...
+
+    Camera {
+        id: camera
+        heading: 0.0
+        pitch: 45.0
+        roll: 0.0
+
+        Point {
+            x: longitude
+            y: latitude - 1.0 // place the camera somewhat southern than the cone
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_QMLSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qml]: > somewhat southern than the cone

should be

> somewhat south of the cone

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/README.md (hunk 1 above)
Comment (mfeigl) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/README.md]: Done.

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/README.md (hunk 2 above)
Comment (mfeigl) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/README.md]: Done.

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.cpp (hunk 4 above)
Comment (mfeigl) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.cpp]: Done.

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.cpp (hunk 5 above)
Comment (mfeigl) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.cpp]: Done.

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.cpp (hunk 6 above)
Comment (mfeigl) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.cpp]: Done.

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.h (hunk 7 above)
Comment (mfeigl) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.h]: Done.

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.pro (hunk 8 above)
Comment (mfeigl) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.pro]: Done.

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qml (hunk 9 above)
Comment (mfeigl) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qml]: Removed.

File: ArcGISRuntimeSDKQt_QMLSamples/Scenes/ScenePropertiesExpressions/README.md (hunk 11 above)
Comment (mfeigl) [reply ArcGISRuntimeSDKQt_QMLSamples/Scenes/ScenePropertiesExpressions/README.md]: Changed.

File: ArcGISRuntimeSDKQt_QMLSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qml (hunk 12 above)
Comment (mfeigl) [reply ArcGISRuntimeSDKQt_QMLSamples/Scenes/ScenePropertiesExpressions/ScenePropertiesExpressions.qml]: Done, added string properties.

**Pull request merged**